### PR TITLE
Javascript/javascript1/week2

### PR DESCRIPTION
**Pull request by Gohar33**

### CODE REVIEW

File: javascript/javascript1/week2/index.js [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+console.log(getEventWeekday(9));
+console.log(getEventWeekday(3));
+
+
+
+//Project 3. Weather wear
+
+function suggestClothes(temp) {
+
+    if (temp < 0) {
```
Comment (florinmaticu) [inline javascript/javascript1/week2/index.js:50]: If you call the function with _0_ then the function returns nothing.
```suggestion
    if (temp <= 0) {
```

File: javascript/javascript1/week2/index.js [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+
+
+
+//Project 4 Student manager
+
+let class07Students = [];
+
+function addStudentToClass(studentName) {
+    if (studentName === "Queen") {
```
Comment (florinmaticu) [inline javascript/javascript1/week2/index.js:73]: If I call your function twice with the name _Queen_ then your code would add her twice. Can you spot the issue ?

File: javascript/javascript1/week2/index.js [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+//Project 5 Candy helper optional
+
+let boughtCandyPrices = [];
+
+
+function addCandy(candyType, weight) {
+    let price = 0;
+    boughtCandyPrices.push(price);
+    if (candyType === "Sweet") {
+        return price = weight * 0, 5;
```
Comment (florinmaticu) [inline javascript/javascript1/week2/index.js:121]: If you return a value but you are not using it then it's not useful.

Maybe you want to do something like 

`boughtCandyPricess.push(addCandy(...))`

Comment (florinmaticu) [reply javascript/javascript1/week2/index.js:121]: Can you post the error in the message ? Thanks

File: javascript/javascript1/week2/index.js [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+
+//Project 4 Student manager
+
+let class07Students = [];
+
+function addStudentToClass(studentName) {
+    if (studentName === "Queen") {
+        class07Students.push(studentName);
+        console.log("We always have a space for our queen")
+    } else if (class07Students.includes(studentName)) {
```
Comment (florinmaticu) [inline javascript/javascript1/week2/index.js:76]: You have to put this check first...before the check for the queen.

File: javascript/javascript1/week2/index.js (hunk 3 above)
Comment (florinmaticu) [reply javascript/javascript1/week2/index.js:121]: Most of the code is good. There are some findings but it's still quite okay.

**Review by florinmaticu — APPROVED** [review]

File: javascript/javascript1/week2/index.js (hunk 2 above)
Comment (florinmaticu) [reply javascript/javascript1/week2/index.js:73]: Yes. Just add the `if` statement that checks for a student in a list before you check for the queen.

Would suggest to use the debugger from visual studio code that can guide step by step and show you what code is executed

File: javascript/javascript1/week2/index.js (hunk 3 above)
Comment (Gohar33) [reply javascript/javascript1/week2/index.js:121]: It says  "Uncaught ReferenceError: boughtCandyPricess is not defined
    at addCandy (index.js:119)
    at index.js:137"

File: javascript/javascript1/week2/index.js (hunk 4 above)
Comment (Gohar33) [reply javascript/javascript1/week2/index.js:76]: Great, thanks, it worked :)

File: javascript/javascript1/week2/index.js (hunk 3 above)
Comment (Gohar33) [reply javascript/javascript1/week2/index.js:121]: Sorry Florin, I don't understand the problem with if part.

Comment (florinmaticu) [reply javascript/javascript1/week2/index.js:121]: boughtCandyPrices**s**  I think you put one **s** more. In your code you have **boughtCandyPrices**

Comment (Gohar33) [reply javascript/javascript1/week2/index.js:121]: It still gives an error, this time something about maximum call stack size exceeded. I changed the location and put it after if statement, the error is gone but it doesn't show anything in the array :/

Comment (florinmaticu) [reply javascript/javascript1/week2/index.js:121]: First you need to remove the array from the function and  `boughtCandyPrices.push()`.
Then I would recommend to do `price = weight *0.5; return price;` you have one assignment and one return

**Pull request merged**